backend/src/api/services/question/question.service.ts

```diff
@@ -41,8 +44,33 @@ export const getSingleQuestionById = async (questionID: string): Promise<AnyQues
 };
 
 export const getSidebarMetadata = async (studentID: string) => {
-    const sidebarMetada: [QuestionMetadata] = await loadData('sidebarQuestions.json');
-    return sidebarMetada;
+    const chapters = await ChapterModel.find();
+    const metadata: SideBarMetadata[] = [];
+    for (const chapter of chapters) {
+        const questions = await QuestionBaseModel.find({ chapter: chapter.questions});
+        for (const question of questions) {
+            const student = await StudentProgressModel.findOne({ studentID });
+
+            if (!student) {
+                throw new HttpError(status.NOT_FOUND, `Student not found with ID ${studentID}`);
+            }
```

Comment:
I feel like this should be outside the for loop. This essentially tries to find a student for every question of every chapter. This logic should be performed once before traversing

Reply:
you are right